iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -231,12 +231,12 @@ private async Task EnsureMessageReceivingLinkAsync(TimeSpan timeout, bool enable
                     {
                         amqpIoTSession.SafeClose();
                     };
-                    if (enableCallback)
-                    {
-                        _messageReceivingLink.RegisterReceiveMessageListener(OnDeviceMessageReceived);
-                    }
                     Logging.Associate(this, this, _messageReceivingLink, nameof(EnsureMessageReceivingLinkAsync));
                 }
+                if (enableCallback)
```

Comment:
newline ahead of this new block, to make it visually distinct from the previous one (unlike try/catch/finally/if/else if/else)